pinot-core/src/main/java/org/apache/pinot/core/realtime/stream/StreamMessageMetadata.java

```diff
@@ -18,24 +18,22 @@
  */
 package org.apache.pinot.core.realtime.stream;
 
-import org.apache.pinot.annotations.InterfaceAudience;
-import org.apache.pinot.annotations.InterfaceStability;
+import org.apache.pinot.common.metadata.RowMetadata;
 
 
 /**
  * A class that provides metadata associated with the message of a stream, for e.g.,
  * ingestion-timestamp of the message.
  */
-@InterfaceAudience.Public
-@InterfaceStability.Evolving
-public class StreamMessageMetadata {
+public class StreamMessageMetadata implements RowMetadata {
 
   private long _ingestionTimestamp = Long.MIN_VALUE;
```

Comment:
Make this final and set it in constructor? I don't think we need to recycle this long value wrapper.